### ISSUE TITLE
[BUG] No native support for auth backends on non-standard paths

### ISSUE THREAD

**Issue by EricHorst** [hashicorp/vault-action#185]
**Issue**
We're using approle and have the approle backend on two separate paths. In setting up vault-action I found that vault-action does not handle alternate paths except with a messy workaround.

The "method" value is overloaded to both indicate the backend type and the path for the backend. That is, when using `method: approle` it is both the indicator to expect the roleId and secretId parameters and implies the path to be "auth/approle/login".  Initially it was not clear to me that this was the case, I assumed method simply meant to expect an approle backend and then I wondered where the path would be specified.

The github auth method suffers from the same limitation.

**Expected behavior**
I would expect that vault-action would support alternate paths to support the flexibility that Vault provides with backends. It would be clearer to have both a `method:` and a `path:` parameter with `path:` defaulting to "approle" for the normies.

**Workaround**
There is enough flexibility in vault-action to work around this shortcoming by using `method: approle/mypath` and providing authPayload thusly:
```              
         with:
            url: https://vault.mycompany.com:8200
            method: approle/mypath
            authPayload: |
                  {
                    "role_id": "${{ secrets.roleId }}",
                    "secret_id": "${{ secrets.secretId }}"
                  } 
```
But this is a messy way to handle what is otherwise straightforward approle auth.

Thanks for considering this bug/feature request.

**Comment by jasonodonnell** [comment]
Supported now in #218.

**Comment by olafz** [comment]
> Supported now in #218.

When will the version be released that includes this fix? The current release (v2.2.0) does not include this fix (yet).

**Comment by tvoran** [comment]
@olafz v2.3.0 was released on June 23rd with this patch.
